pandas/indexes/frozen.py

```diff
@@ -25,11 +27,12 @@ class FrozenList(PandasObject, list):
     #           typechecks
 
     def __add__(self, other):
-        if isinstance(other, tuple):
-            other = list(other)
-        return self.__class__(super(FrozenList, self).__add__(other))
+        warnings.warn("__add__ is deprecated, use union(...)", FutureWarning)
+        return self.union(other)
 
-    __iadd__ = __add__
+    def __iadd__(self, other):
+        warnings.warn("__iadd__ is deprecated, use union(...)", FutureWarning)
+        return self.union(other)
```

Comment:
this is technically wrong; its an in-place modify, and doesn't return anything. Can you add a test for validate this?

*maybe*

```
def __iadd__(self, other):
    warnings(.....)
     if isinstance(other, tuple):
            other = list(other)
    super(FrozenList, self).__iadd__(other)
```

might work